ledger/index.ts

```diff
@@ -521,41 +521,45 @@ export async function signSimpleBip322Message({
 /**
  * This function is used to get the stx account data from the ledger
  * @param transport - the transport object with connected ledger device
+ * @param network - the network type (Mainnet or Testnet)
+ * @param accountIndex - the account index of the account to sign with
+ * @param addressIndex - the index of the account address to sign with
+ * @param showAddress - show address on the wallet's screen
  * @returns the address and the public key in compressed format
  * */
 export async function importStacksAccountFromLedger(
   transport: Transport,
   network: NetworkType,
   accountIndex = 0,
   addressIndex = 0,
+  showAddress = false,
 ): Promise<{ address: string; publicKey: string }> {
   const appStacks = new StacksApp(transport);
-
-  const { address, publicKey } = await appStacks.getAddressAndPubKey(
-    `m/44'/5757'/${accountIndex}'/0/${addressIndex}`,
-    network === 'Mainnet' ? AddressVersion.MainnetSingleSig : AddressVersion.TestnetSingleSig,
-  );
+  const path = `m/44'/5757'/${accountIndex}'/0/${addressIndex}`;
+  const version = network === 'Mainnet' ? AddressVersion.MainnetSingleSig : AddressVersion.TestnetSingleSig;
+  const { address, publicKey } = showAddress
+    ? await appStacks.showAddressAndPubKey(path, version)
+    : await appStacks.getAddressAndPubKey(path, version);
 
   return { address, publicKey: publicKey.toString('hex') };
 }
 
 /**
  * This function is used to sign a Stacks transaction with the ledger
  * @param transport - the transport object with connected ledger device
- * @param transaction - the transaction to sign
+ * @param transactionBuffer - the transaction to sign
```

Comment:
was there a strong reason to change this interface and move the serialize call out of this function?

I noticed your web-extension PR didn't update confirmLedgerTransaction/index.ts L251 with the new param

and confirmStxTransaction/index.ts now does an extra .seralize() before calling this function

Reply:
Yeah, the serialization was added on the extension side because otherwise it threw an error:
<img width="635" alt="Monosnap Xverse Wallet 2023-09-01 13-13-16" src="https://github.com/secretkeylabs/xverse-core/assets/36603049/58657c6b-faf6-4ee7-b043-4a8afc1d349f">

Currently, `confirmStxTransaction/index.ts` makes the tx serialization and passes it to `confirmLedgerTransaction/index.ts` screen ready to be passed to this core function in a Buffer format

Reply:
right. basically because we have to pass around a serialized transaction between screens anyway, so this removes the multiple actions to serialize (for location.state) -> deserialize (for this function) -> serialize (within this function)

is that right?

Reply:
Yeah, that's right